cnd/src/network/mod.rs

```diff
@@ -728,16 +731,16 @@ async fn insert_state_for_bob<AL, BL, AA, BA, AH, BH, AI, BI, AT, BT, DB>(
     swap_request: Request<AL, BL, AA, BA, AI, BI>,
 ) -> anyhow::Result<()>
 where
-    AL: Send + 'static,
-    BL: Send + 'static,
-    AA: Send + 'static,
-    BA: Send + 'static,
-    AH: Send + 'static,
-    BH: Send + 'static,
-    AI: Send + 'static,
-    BI: Send + 'static,
-    AT: Send + 'static,
-    BT: Send + 'static,
+    AL: Clone + Send + Sync + 'static,
+    BL: Clone + Send + Sync + 'static,
+    AA: Clone + Ord + Send + Sync + 'static,
+    BA: Clone + Ord + Send + Sync + 'static,
+    AH: Clone + Send + Sync + 'static,
+    BH: Clone + Send + Sync + 'static,
+    AI: Clone + Send + Sync + 'static,
+    BI: Clone + Send + Sync + 'static,
+    AT: Clone + Send + Sync + 'static,
+    BT: Clone + Send + Sync + 'static,
     DB: Save<Request<AL, BL, AA, BA, AI, BI>> + Save<Swap>,
```

Comment:
```suggestion
	SwapEvent<AA, BA, AH, BH, AT, BT>: Clone + Sync 
	...
```
Where does the trait bound requirement come from?
Would the following work instead?

Reply:
Almost! It's

```rust
bob::State<AL, BL, AA, BA, AH, BH, AI, BI, AT, BT>: Clone + Sync,
```

Thank you!